microsoft-edge/webview2/how-to/winrt-from-js.md

```diff
@@ -343,8 +353,34 @@ Next, use the DevTools Console to demonstrate that web-side code can call the in
    ![Using the DevTools Console to test calling native-side code from web-side code.](winrt-from-js-images/devtools-console-calling-native-side-code.png)
 
 Congratulations!  You've finished the sample demonstration of calling WinRT code from JavaScript code.
+## Custom (3rd Party) WinRT Components
+
+The WebView2 WinRT tool supports custom WinRT component projections as long as its using a C# base WinRT class.
+
+The following are the steps you will need to take:
+
+1. Add a third project (other than your main app and WinRTAdapter project) to your VS solution that implements your winrt class.
+2. Have the WinRTAdapter project 'Add a reference' to your new third project containing your winrt class.
+3. Update the WinRTAdapter project's Include filter in the properties to also include your new class.
+4. Add an additional line to InitializeWebView2Async to add your winrt class's namespace `WebView2.CoreWebView2.AddHostObjectToScript("MyCustomNamespace", dispatchAdapter.WrapNamedObject("MyCustomNamespace", dispatchAdapter));`
+1. For easy method calling from the web, optionally add your namespace sync proxy as a global object in script. Ex. `window.MyCustomNamespace = chrome.webview.hostObjects.sync.MyCustomNamespace;`
 
+See the following [sample app](https://github.com/MicrosoftEdge/WebView2Samples/compare/uwp-wv2winrt-custom-csharp-winrt) to see an example of this.
+## Asynchronous calls into WinRT
 
+Following the steps in the above guide you should be able to use synchronous proxies. For async method calls you will need to use chrome.webview.hostObjects.options.forceAsyncMethodMatches. 
+
+The property [forceAsyncMethodMatches](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2?view=webview2-winrt-1.0.1264.42#addhostobjecttoscript) is an array of regex that if any match a method name on a sync proxy, the method will be run asynchronously instead. Setting this to [/Async$/] will have it match any method ending with the suffix Async. Then matching method calls work just like a method on an async proxy and returns a promise that you can await.
```

Comment:
```suggestion
The property [`forceAsyncMethodMatches`](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2?view=webview2-winrt-1.0.1264.42#addhostobjecttoscript) is an array of regex that if any match a method name on a sync proxy, the method will be run asynchronously instead. Setting this to `[/Async$/]` will have it match any method ending with the suffix `Async`. Then matching method calls work just like a method on an async proxy and returns a promise that you can await.
```

Reply:
fyi i tweaked David's Suggestion to remove url prefix & ?view param

Reply:
handling in my png commit